Re-implement get_commit_data in kernel_tree

fgyanz opened the pull request:
And adapt the CI tests to use kernel.git

fgyanz committed:
klplib/kernel-tree: re-implement get_commit_data

Downloading the upstream patches from 'kernel.org' has
become unstable and unreliable for the last months.
So, instead, fetch the commits' data from the local
kernel.git repository.

Signed-off-by: Fernando Gonzalez <fernando.gonzalez@suse.com>

fgyanz committed:
tests: Adapt the tests to use kernel.git repo

Signed-off-by: Fernando Gonzalez <fernando.gonzalez@suse.com>

vmezzela reviewed (commented):
As mentioned [here](https://github.com/SUSE/klp-build/pull/110#issuecomment-2793987889), we should update the kernel tree before using it. If you plan to merge this PR before the other, please do it here and forget the comment there.

Besides that, one less call to an external service is a good thing. I'm testing it :)

fgyanz committed:
plugins/scan: Update kernel trees on scan

The update was done solely on 'setup' and 'extract' commands,
as they were the only ones accessing the kernel repository.
From now on, as with the kernel-sources, the kernel trees
will also get updated when running scan too.

Signed-off-by: Fernando Gonzalez <fernando.gonzalez@suse.com>

fgyanz commented on klpbuild/klplib/kernel_tree.py:
```diff
@@ -8,10 +8,32 @@
 from klpbuild.klplib.utils import ARCH
 
 
+# Currently this function returns the date of the patch and it's subject
```
Oops copy-pasting typo. Thanks!

fgyanz commented:
Done. Thanks @vmezzela

vmezzela reviewed (approved):
lgtm!

marcosps reviewed (approved):
Tested, LGTM!
